libraries/helpers.rb

```diff
@@ -51,6 +51,28 @@ def ingredient_ctl_command
     end
 
     #
+    # Returns license acceptance CLI parameter when applicable
+    #
+    def license_acceptance_parameter
+      if new_resource.accept_license &&
+         (new_resource.product_name == 'compliance' && version_for(new_resource.version) >= version_for('1.1.3')) ||
+         (new_resource.product_name == 'manage' && version_for(new_resource.version) >= version_for('1.10.0')) ||
+         (new_resource.product_name == 'reporting' && version_for(new_resource.version) >= version_for('1.6.0')) ||
+         (new_resource.product_name == 'analytics' && version_for(new_resource.version) >= version_for('1.4.0'))
+        '--accept_license'
```

Comment:
So IMO we should just lay down the required `/var/opt/<PROJECT>/.license.accepted` if `accept_license` is true. This will allow us to avoid all this crazy version checking.